src/store/api/gateway/proposers.ts

```diff
@@ -15,18 +15,29 @@ export const proposerEndpoints = (
   }),
   deleteProposer: builder.mutation<
     void,
-    { chainId: string; safeAddress: string; delegateAddress: string; signature: string }
+    {
+      chainId: string
+      safeAddress: string
+      delegateAddress: string
+      delegator: string
+      signature: string
+    }
   >({
-    queryFn({ chainId, safeAddress, delegateAddress, signature }) {
+    queryFn({ chainId, safeAddress, delegateAddress, delegator, signature }) {
       return buildQueryFn(() =>
-        deleteDelegateV2({ params: { path: { chainId, delegateAddress } }, body: { safe: safeAddress, signature } }),
+        deleteDelegateV2({
+          params: { path: { chainId, delegateAddress } },
+          body: { safe: safeAddress, signature, delegator },
+        }),
       )
     },
     // Optimistically update the cache and roll back in case the mutation fails
-    async onQueryStarted({ chainId, safeAddress, delegateAddress }, { dispatch, queryFulfilled }) {
+    async onQueryStarted({ chainId, safeAddress, delegateAddress, delegator }, { dispatch, queryFulfilled }) {
```

Comment:
This is cool. Just wondering - can we use cache tags here as well and once the query is complete also invalidate the cache for the getDelegates query?

Reply:
I think it is possible to use tags here as well to do a re-fetch whenever a proposer is added or deleted. However, to avoid too many requests until our [backend is optimized](https://www.notion.so/safe-global/Cache-Proposers-data-in-the-CGW-1308180fe57380c2a524f3e50c68ef23) I followed this guide of optimistically updating the cache: https://redux-toolkit.js.org/rtk-query/usage/manual-cache-updates#optimistic-updates